src/viam/sdk/services/mlmodel/client.cpp

```diff
@@ -48,6 +48,9 @@ std::shared_ptr<MLModelService::named_tensor_views> MLModelServiceClient::infer(
     auto& mutable_input_data = *req->mutable_input_data();
     auto& mutable_input_data_fields = *mutable_input_data.mutable_fields();
 
+    // TODO: Currently, this doesn't validate that we are passing the
```

Comment:
I agree that after the flat tensors are added, this will not be necessary- the type check can always be added in later if needed or if flat tensors turn out not to be viable.